runtime/src/signed_extensions/check_call_allowed.rs

```diff
@@ -0,0 +1,148 @@
+    // otherwise returns true.
+    fn has_no_invalid_bonding_calls(who: &AccountId, calls: Vec<Call>) -> bool {
+        let mut all_calls_valid = true;
+
+        for call in calls.into_iter() {
+            all_calls_valid = all_calls_valid
+                && match call {
+                    // Calls that can contain other Calls and must be checked recursively..
+                    Call::Utility(substrate_utility::Call::<Runtime>::batch { calls }) => {
+                        Self::has_no_invalid_bonding_calls(who, calls.to_vec())
+                    }
+                    Call::Utility(substrate_utility::Call::<Runtime>::as_derivative {
+                        call,
+                        ..
+                    }) => Self::has_no_invalid_bonding_calls(who, vec![*call]),
+
+                    // Bonding
+                    Call::Staking(pallet_staking::Call::<Runtime>::bond { .. }) => {
+                        Self::has_no_conflicting_locks(who)
+                    }
+
+                    // should we prevent Sudo from bypassing these checks?
+                    // Call::Sudo(pallet_sudo::Call::<Runtime>::sudo { call }) => ...
+                    // Call::Sudo(pallet_sudo::Call::<Runtime>::sudo_as { who, call }) => ...
+                    // Call::Sudo(pallet_sudo::Call::<Runtime>::sudo_unchecked_weight { call }) => ...
+                    // Call::Utility(substrate_utility::Call::<Runtime>::batch_all { calls }) => ...
+                    // Call::Utility(substrate_utility::Call::<Runtime>::force_batch { calls }) => ...
+                    // Call::Utility(substrate_utility::Call::<Runtime>::dispatch_as { calls }) => ...
+                    _ => true,
+                };
+
+            if !all_calls_valid {
+                return false;
+            }
+        }
+
+        all_calls_valid
+    }
+}
+impl SignedExtension for CheckCallAllowed<Runtime> {
+    type AccountId = AccountId;
+    type Call = Call;
+    type AdditionalSigned = ();
+    type Pre = ();
+    const IDENTIFIER: &'static str = "CheckCallAllowed";
+
+    fn additional_signed(&self) -> sp_std::result::Result<(), TransactionValidityError> {
+        Ok(())
+    }
+
+    fn pre_dispatch(
+        self,
+        who: &Self::AccountId,
+        call: &Self::Call,
+        info: &DispatchInfoOf<Self::Call>,
+        len: usize,
+    ) -> Result<Self::Pre, TransactionValidityError> {
+        self.validate(who, call, info, len).map(|_| ())
+    }
+
+    fn validate(
+        &self,
+        who: &Self::AccountId,
+        call: &Self::Call,
+        _info: &DispatchInfoOf<Self::Call>,
+        _len: usize,
+    ) -> TransactionValidity {
+        if Self::has_no_invalid_bonding_calls(who, vec![call.clone()]) {
+            Ok(ValidTransaction {
+                priority: 0,
+                requires: vec![],
+                provides: vec![],
+                longevity: TransactionLongevity::max_value(),
+                propagate: true,
+            })
```

Comment:
It seems to be the same as:
```suggestion
            Ok(Default::default())
```
(which is returned by `frame_system::CheckWeight` for example)